actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -40,6 +39,12 @@ def call(env)
         # CSP headers might not match nonces in the cached HTML.
         return response if status == 304
 
+        # Prior to Rack 3, headers are not normalized to lowercase.
+        # Apps using Rack 2.x may have set the CSP header with a capitalized name.
+        # Using Rack::Headers ensures that the header name is normalized.
+        # This can be removed when Rack 2.x is no longer supported.
+        headers = Rack::Headers[headers]
+        response[1] = headers
         return response if policy_present?(headers)
```

Comment:
What do we think about wrapping the whole response in `Rack::Response` instead of setting the headers specifically, ie. `response = Rack::Response[*response]`?

Reply:
Let's go with conditional casing for now, for example: https://github.com/rails/rails/commit/b580df616b1d6d32cc7dee1505f8ae84d0444e04#diff-949c826f80ade38bff4ede16c161fa0da4f24068a8e9df914e50e0a931516850R53-R60

This should prevent the need to use `Rack::Response`/`Rack::Headers`

Reply:
I've added ActionDispatch::Constants (to mirror constants under `Rack::` namespace) to avoid spreading the conditionals around multiple file:

https://github.com/rails/rails/commit/1fd79abb549739c3db06f9fe197ff8836be639c0

Reply:
Thanks. I've updated and rebased this branch.